lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -20,15 +20,187 @@ const getMethodAuthorization = require('./lib/method/authorization');
+    oneOf: [
+      { type: 'boolean' },
+      {
+        type: 'object',
+        properties: {
+          required: { type: 'boolean' },
+          mappedValue: { type: 'string' },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+};
+
+const authorizerSchema = {
+  oneOf: [
+    { type: 'string' },
+    caseInsensitive('aws_iam'),
+    { $ref: '#/definitions/awsArnString' },
+    {
+      type: 'object',
+      properties: {
+        arn: { $ref: '#/definitions/awsArn' },
+        authorizerId: { $ref: '#/definitions/awsCfInstruction' },
+        claims: { type: 'array', items: { type: 'string' } },
+        identitySource: { type: 'string' },
+        identityValidationExpression: { type: 'string' },
+        managedExternally: { type: 'boolean' },
+        name: { type: 'string' },
+        resultTtlInSeconds: { type: 'integer', minimum: 0, maximum: 3600 },
+        scopes: { type: 'array', items: { type: 'string' } },
+        type: {
+          oneOf: ['token', 'cognito_user_pools', 'request', 'aws_iam'].map(caseInsensitive),
+        },
+      },
+      required: [],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const corsSchema = {
+  oneOf: [
+    { type: 'boolean' },
+    {
+      type: 'object',
+      properties: {
+        allowCredentials: { type: 'boolean' },
+        cacheControl: { type: 'string' },
+        headers: { type: 'array', items: { type: 'string' } },
+        maxAge: { type: 'integer', minimum: 1 },
+        methods: { type: 'array', items: { enum: allowedMethods } },
+      },
+      oneOf: [
+        { properties: { origin: { type: 'string' } } },
+        {
+          properties: {
+            origins: {
+              type: 'array',
+              items: { type: 'string' },
+            },
+          },
+        },
+      ],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const requestSchema = {
+  type: 'object',
+  properties: {
+    contentHandling: contentHandlingSchema,
+    method: { type: 'string', regexp: methodPattern.toString() },
+    parameters: {
+      type: 'object',
+      properties: {
+        querystrings: requestParametersSchema,
+        headers: requestParametersSchema,
+        paths: requestParametersSchema,
+      },
+      additionalProperties: false,
+    },
+    passThrough: { enum: ['NEVER', 'WHEN_NO_MATCH', 'WHEN_NO_TEMPLATES'] },
+    schema: {
+      type: 'object',
+      additionalProperties: { type: ['object', 'string'] },
+    },
+    template: {
+      type: 'object',
+      additionalProperties: { type: ['string', 'null'] },
+    },
+    uri: { type: 'string' },
+  },
+  additionalProperties: false,
+};
+
+const responseSchema = {
+  type: 'object',
+  properties: {
+    contentHandling: contentHandlingSchema,
+    headers: {
+      type: 'object',
+      additionalProperties: { type: 'string' },
+    },
+    template: { type: 'string' },
+    statusCodes: {
+      type: 'object',
+      additionalProperties: {
```

Comment:
It'll be nice to restrict property names so they can only follow pattern of `\d{3}`